sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -116,47 +116,53 @@ object OrcUtils extends Logging {
   }
 
   /**
-   * Returns the requested column ids from the given ORC file. Column id can be -1, which means the
-   * requested column doesn't exist in the ORC file. Returns None if the given ORC file is empty.
+   * @return Returns the requested column ids from the given ORC file and Boolean flag to use actual
+   * schema or result schema. Column id can be -1, which means the requested column doesn't
+   * exist in the ORC file. Returns None if the given ORC file is empty.
    */
   def requestedColumnIds(
       isCaseSensitive: Boolean,
       dataSchema: StructType,
       requiredSchema: StructType,
       reader: Reader,
-      conf: Configuration): Option[Array[Int]] = {
+      conf: Configuration): (Option[Array[Int]], Boolean) = {
+    var sendActualSchema = false
     val orcFieldNames = reader.getSchema.getFieldNames.asScala
     if (orcFieldNames.isEmpty) {
```

Comment:
If we can't do column pruning for the _col case, shall we simply return None? e.g.
```
if (orcFieldNames.isEmpty || orcFieldNames.forall(_.startsWith("_col"))) {
  None
}
```

Reply:
This will not give correct result, On changing the above comment, results are showing empty rows

```
scala> val u = """select date_dim.d_date_id from date_dim limit 5"""
u: String = select date_dim.d_date_id from date_dim limit 5

scala> spark.sql(u).collect
res1: Array[org.apache.spark.sql.Row] = Array()

scala> val u = """select * from date_dim limit 5"""
u: String = select * from date_dim limit 5
scala> spark.sql(u).collect
res2: Array[org.apache.spark.sql.Row] = Array()
```